### PR TITLE
Don't leak watch files

### PR DESCRIPTION
**Pull request by divolgin**
What I Did
------------

Fixed a temp file leak in watch.

How I Did it
------------

Delete the temp folder with git repo contents.

How to verify it
------------

Follow steps to reproduce in #762 

Description for the Changelog
------------

Fixed file leak when running `ship watch`.

Picture of a Boat (not required but encouraged)
------------

leak ship

### CODE REVIEW

File: pkg/specs/interface_test.go [hunk 1]
```diff
@@ -209,18 +221,21 @@ icon: https://kfbr.392/x5.png
...
+				localApp *apptype.MockLocalAppCopy,
 				mockState *state.MockManager,
 				mockFs afero.Afero,
 				mockAppResolver *replicatedapp.MockResolver,
 				mockGitHubFetcher *githubclient.MockGitHubFetcher,
 			) {
 				inOrder := mockUi.EXPECT().Info("Reading /path/to/ship.yaml?customer_id=123&installation_id=456&release_semver=789 ...")
 				inOrder = mockUi.EXPECT().Info("Determining application type ...").After(inOrder)
-				inOrder = appType.EXPECT().
+				inOrder = inspector.EXPECT().
```
Comment (dexhorthy) [inline pkg/specs/interface_test.go]: nice

File: pkg/specs/interface.go [hunk 2]
```diff
@@ -210,49 +207,48 @@ func (r *Resolver) ResolveRelease(ctx context.Context, upstream string) (*api.Re
...
+	app, err := r.appTypeInspector.DetermineApplicationType(ctx, upstream)
 	if err != nil {
 		return "", errors.Wrapf(err, "resolve app type for %s", upstream)
 	}
-	debug.Log("event", "apptype.inspect", "type", appType, "localPath", localPath)
+	debug.Log("event", "apptype.inspect", "type", app.GetType(), "localPath", app.GetLocalPath())
 
 	defer func() {
-		if err := r.FS.RemoveAll(localPath); err != nil {
+		if err := app.Remove(r.FS); err != nil {
```
Comment (dexhorthy) [inline pkg/specs/interface.go]: what is the reasoning behind passing in the FS here? I would have expected to just see `localPath` replaced with `app.GetLocalPath()` everywhere (which is a really solid refactor IMO). what does this get us over `r.FS.RemoveAll(app.GetLocalPath)`

Comment (dexhorthy) [reply pkg/specs/interface.go]: (not saying this is a blocker, just want to understand the motivations, and what problems you may have hit that I'm not thinking of)

Comment (divolgin) [reply pkg/specs/interface.go]: @dexhorthy `localPath` is the location of the chart within the cloned repo, but we need to remove the entire temp folder, which is some arbitrary number of directories up in the file tree.

Comment (divolgin) [reply pkg/specs/interface.go]: If you look in that `Remove` function, it doesn't remove localPath but a different path.

Comment (dexhorthy) [reply pkg/specs/interface.go]: gotchaaa okay makes sense. Thanks for clarifying

**Pull request merged**